crates/types/src/vid.rs

```diff
@@ -0,0 +1,254 @@
+    vid::{
+        advz::{
+            payload_prover::{LargeRangeProof, SmallRangeProof},
+            Advz,
+        },
+        payload_prover::{PayloadProver, Statement},
+        precomputable::Precomputable,
+        VidDisperse, VidResult, VidScheme,
+    },
+};
+use serde::{Deserialize, Serialize};
+use sha2::Sha256;
+use std::{fmt::Debug, ops::Range};
+
+/// VID scheme constructor.
+///
+/// Returns an opaque type that impls jellyfish traits:
+/// [`VidScheme`], [`PayloadProver`], [`Precomputable`].
+///
+/// # Rust forbids naming impl Trait in return types
+///
+/// Due to Rust limitations the return type of [`vid_scheme`] is a newtype
+/// wrapper [`VidSchemeType`] that impls the above traits.
+///
+/// We prefer that the return type of [`vid_scheme`] be `impl Trait` for the
+/// above traits. But the ability to name an impl Trait return type is
+/// currently missing from Rust:
+/// - [Naming impl trait in return types - Impl trait initiative](https://rust-lang.github.io/impl-trait-initiative/explainer/rpit_names.html)
+/// - [RFC: Type alias impl trait (TAIT)](https://github.com/rust-lang/rfcs/blob/master/text/2515-type_alias_impl_trait.md)
+///
+/// # Panics
+/// When the construction fails for the underlying VID scheme.
+#[must_use]
+pub fn vid_scheme(num_storage_nodes: usize) -> VidSchemeType {
+    // TODO use a proper SRS
+    // https://github.com/EspressoSystems/HotShot/issues/1686
+    let srs = {
+        let mut rng = jf_utils::test_rng();
+        UnivariateKzgPCS::<E>::gen_srs_for_testing(
+            &mut rng,
+            checked_fft_size(num_storage_nodes).unwrap(),
+        )
+        .unwrap()
+    };
+
+    // chunk_size is currently num_storage_nodes rounded down to a power of two
+    // TODO chunk_size should be a function of the desired erasure code rate
+    // https://github.com/EspressoSystems/HotShot/issues/2152
+    let chunk_size = 1 << num_storage_nodes.ilog2();
+
+    // TODO intelligent choice of multiplicity
+    let multiplicity = 1;
+
+    // TODO panic, return `Result`, or make `new` infallible upstream (eg. by panicking)?
+    #[allow(clippy::panic)]
+    VidSchemeType(Advz::new(chunk_size, num_storage_nodes, multiplicity, srs).unwrap_or_else(|err| panic!("advz construction failure:\n\t(num_storage nodes,chunk_size,multiplicity)=({num_storage_nodes},{chunk_size},{multiplicity})\n\terror: : {err}")))
+}
+
+/// VID commitment type
+pub type VidCommitment = <VidSchemeType as VidScheme>::Commit;
+/// VID common type
+pub type VidCommon = <VidSchemeType as VidScheme>::Common;
+/// VID share type
+pub type VidShare = <VidSchemeType as VidScheme>::Share;
+
+/// Newtype wrapper for a VID scheme type that impls
+/// [`VidScheme`], [`PayloadProver`], [`Precomputable`].
+pub struct VidSchemeType(Advz<E, H>);
+
+/// Newtype wrapper for a namespace proof.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct NamespaceProofType(
```

Comment:
Nit: other than here, HotShot doesn't know about namespaces at all (and we're even mostly removing its knowledge of transactions). It might make more sense to still call these `{Large,Small}RangeProof`, just without the type parameters.

Reply:
Ooh good catch thank you. Fixed in https://github.com/EspressoSystems/HotShot/pull/2670/commits/195edec572bb013552280d1a3f6371a490d8251f